### ISSUE TITLE
ruff format with quote-style = "preserve" should not show ICS001 and flake8-quotes.multiline-quotes warnings

### ISSUE THREAD

**Issue by 57an** [astral-sh/ruff#11063]
While running ruff format with quote-style = "preserve" option I get such warnings

```
warning: The following rules may cause conflicts when used with the formatter: `ISC001`. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding them to the `ignore` configuration.
warning: The `flake8-quotes.multiline-quotes="single"` option is incompatible with the formatter. We recommend disabling `Q001` when using the formatter, which enforces double quotes for multiline strings. Alternatively, set the `flake8-quotes.multiline-quotes` option to `"double"`.`
```

I think with quote-style = "preserve" options ruff should not show this warnings, because this option forces string quotes format to be unchanged.

**Comment by MichaReiser** [comment]
I agree that the `flake8-quotes.multiline-quotes` should not be shown. However, the option `quote-style: "preserve"` does not resolve the incompatibility to `ISC001` as you can see here https://play.ruff.rs/569c1110-6903-4622-8694-157b243ea73d
